src/mc-range-edge-generation.c

```diff
@@ -23,29 +23,42 @@
 
 struct _mc_edges_t {
     size_t sparsity;
+    uint32_t trimFactor;
     /* edges is an array of `char*` edge strings. */
     mc_array_t edges;
 };
 
-static mc_edges_t *mc_edges_new(const char *leaf, size_t sparsity, mongocrypt_status_t *status) {
+static mc_edges_t *mc_edges_new(const char *leaf, size_t sparsity, uint32_t trimFactor, mongocrypt_status_t *status) {
     BSON_ASSERT_PARAM(leaf);
     if (sparsity < 1) {
         CLIENT_ERR("sparsity must be 1 or larger");
         return NULL;
     }
+
+    const size_t leaf_len = strlen(leaf);
+    if (trimFactor != 0 && trimFactor >= leaf_len) {
+        // We append a total of leaf_len + 1 (for the root) - trimFactor edges. When this number is equal to 1, we
+        // degenerate into equality, which is not desired, so trimFactor must be less than leaf_len.
```

Comment:
This is true; however, even without setting the trim factor > 0 here, this pretty much ends up being a degenerate case since we only have the root node + leaves. I think that we should probably consider re-evaluating having a maximum value set for sparsity. I'll get some feedback on what people think on this. Thank you for pointing this out.